client-runtime/src/main/java/com/microsoft/rest/v2/http/NettyClient.java

```diff
@@ -371,6 +363,7 @@ void onError(Throwable cause) {
         private ResponseContentFlowable contentEmitter;
         private Subscription requestContentSubscription;
         private final NettyAdapter adapter;
+        private boolean didEmitHttpResponse;
```

Comment:
Does this need to be atomic?

Reply:
I think no, because the purpose of this is for the "happy path", to prevent the .dispose() after completion of the Single from closing the channel before the HttpContents can be read. But we control the timing for that dispose() because we call .onSuccess():

https://github.com/RikkiGibson/autorest-clientruntime-for-java/blob/44be63a9106b7b2b965d70f6341ca64115134df8/client-runtime/src/main/java/com/microsoft/rest/v2/http/NettyClient.java#L426

Reply:
sounds good.